spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/config/DataFlowClientProperties.java

```diff
@@ -0,0 +1,76 @@
+package org.springframework.cloud.dataflow.rest.client.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.dataflow.core.DataFlowPropertyKeys;
+
+/**
+ * Configuration properties used in {@link DataFlowClientAutoConfiguration}
+ *
+ * @author Vinicius Carvalho
+ */
+@ConfigurationProperties(prefix = DataFlowPropertyKeys.PREFIX + "client")
+public class DataFlowClientProperties {
+
+	private String target = "http://localhost:9393";
+
+	private Security security = new Security();
+
+	private boolean skipSslValidation = true;
+
+	private boolean enableDsl = false;
+
+	public boolean isEnableDsl() {
+		return enableDsl;
+	}
+
+	public void setEnableDsl(boolean enableDsl) {
+		this.enableDsl = enableDsl;
+	}
+
+	public boolean isSkipSslValidation() {
+		return skipSslValidation;
+	}
+
+	public void setSkipSslValidation(boolean skipSslValidation) {
+		this.skipSslValidation = skipSslValidation;
+	}
+
+	public String getTarget() {
+		return target;
+	}
+
+	public void setTarget(String target) {
+		this.target = target;
+	}
+
+	public Security getSecurity() {
+		return security;
+	}
+
+	public void setSecurity(Security security) {
+		this.security = security;
+	}
+
+	public static class Security {
```

Comment:
This is too generic a name for basic auth.  There was some careful thought into redoing spring boot security autoconfiguration, we can follow their lead and use `User` as the class to specify username and password for basic auth.

https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SecurityProperties.java#L107

@ghillert @ilayaperumalg What do you think about including ROLES in the User class?

Reply:
Wasn't `security` the name of the property from boot? I have no problem in changing that, but I used boot as a reference (at least 1.x branch that is)

Reply:
Yes, along with the username/password, we need the option to specify `role` as a list of accepted values (VIEW, CREATE, MANAGE). This is in line with what Boot does [here](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SecurityProperties.java#L122)

Reply:
Spring Boot 1.5.x has this https://github.com/spring-projects/spring-boot/blob/1.5.x/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SecurityProperties.java

The reference docs have examples like you coded it, but they were just examples.  Note that the 1.5.x branch doesn't have the roles.  Seems like moving to the way that boot 2 does things will help.

Reply:
A nested class 'Basic' inside Security would make room for oauth2 etc.

Also your suggestion to rename 'security' to authentication is a good one.